src/handlers/psp.ts

```diff
@@ -56,6 +56,27 @@ export const handler = async (_event: any = {}): Promise<any> => {
   return { statusCode: 200, body: JSON.stringify(response) };
 };
 
+const groupProviders = (config: Config) => {
+  const evmChains = config.chains.filter((chain) => chain.type === 'evm');
+  if (isEmpty(evmChains)) {
+    throw new Error('One or more evm compatible chains must be defined in the provided config');
+  }
+  const groupedProviders = evmChains.flatMap((chain) =>
+    Object.entries(chain.providers).map(([providerName, chainProvider]) => {
+      return {
+        chainId: chain.id,
+        providerName,
+        providerUrl: chainProvider.url,
+        chainConfig: chain,
+      };
+    })
+  );
+
+  const filteredGroupedProviders = groupedProviders.filter((p) => !isNil(p));
+
+  return filteredGroupedProviders;
+};
```

Comment:
We don't need to initialize providers in the main function anymore, so I simply converted the function to group them with the necessary data.